core/router/gateway_status.go

```diff
@@ -21,11 +20,5 @@ func (r *router) HandleGatewayStatus(gatewayID string, status *pb_gateway.Status
 		}
 	}()
 
-	gateway := r.getGateway(gatewayID)
-	gateway.LastSeen = time.Now()
-	err = gateway.Status.Update(status)
-	if err != nil {
-		return err
-	}
-	return nil
+	return r.getGateway(gatewayID).HandleStatus(status)
```

Comment:
it is set on return, isn't it?
how is this situation different from the one in the playground example?